framework/src/userobject/ElementUserObject.C

```diff
@@ -52,3 +52,8 @@ ElementUserObject::ElementUserObject(const InputParameters & parameters) :
     addMooseVariableDependency(coupled_vars[i]);
 }
 
+bool
+ElementUserObject::requestedJacobian(unsigned int var) const
+{
+  return var < _requested_jacobian_flag.size() && _requested_jacobian_flag[var];
```

Comment:
I do think the in loop checking of each one is unnecessary... but it may still be a decent plan to form a list of only the variables that you need UserObject Jacobians for.

There could be quite a LOT of time spent calling `prepareShapes()`, etc. for all variables in the system when we really don't need to.